app/assets/javascripts/components/instructor_panel.js.jsx

```diff
@@ -7,6 +7,20 @@ var InstructorPanel = React.createClass({
       action: this.props.queuePop,
     };
   },
+  getQueuePopandPinButtonData: function () {
+    index = 0;
+    for(var r=0; r< this.props.requests.length; r++){
```

Comment:
I used that function to determine how many unpinned requests were left , but kept some of that logic to get the right request, because it is able to retrieve more useful information than getUnpinnedRequests.